plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -174,10 +174,16 @@
      * The OpenShift deploy timeout in seconds:
      * See this issue for background of why for end users on slow wifi on their laptops
      * DeploymentConfigs usually barf: https://github.com/openshift/origin/issues/10531
+     *
+     * Lets default to 2 hours which should be plenty of time on a slow connection to download
+     * all the images for a YAML/JSON manifest (which can contain many separate images)
      */
-    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds")
+    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds", defaultValue = "7200")
```

Comment:
I find it really unfortunate this timeout value is part of the `DeploymentConfig` so that we need to mangle with this parameter on given resource objects (which can come from dependencies, too). 

Following the upstream issues https://github.com/kubernetes/kubernetes/issues/3074, https://github.com/openshift/origin/issues/10531 and https://github.com/openshift/origin/issues/7522 it seems that there is some (but slow) progress in separating image download times and the other deployment startup. 

I wonder whether the referenced `ProgressDeadlineSconds` API can help here (have to look it up) or whether we should switch to plain `Deployment` as soon as they land in OpenShift (as it is supposed to be planned for a future version).

I'm d'accord with increasing the timeout (maybe 1h is enough for now ?) considering to remove the default at any time when finding a better support from the upstream APIs.